src/components/People/PeopleTable.tsx

```diff
@@ -0,0 +1,228 @@
+import type { Person } from '../../types';
+import { TitleTableHeaders } from '../../types';
+import { getPreparedPeople } from '../../utils/getPreparedPeople';
+import { getSearchWith } from '../../utils/searchHelper';
+import { Loader } from '../Loader';
+import { PeopleFilters } from './PeopleFilters';
+import { PersonItem } from './PersonItem';
+import { SearchLink } from '../SearchLink';
+
+const DESCENDING_ORDER = 'desc';
+
+export const PeopleTable = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(person => {
+        setPeople(person);
+      })
+      .catch(() => {
+        setIsError(true);
+        setTimeout(() => setIsError(false), 3000);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  const sex = searchParams.get('sex') || null;
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const showTable = !isLoading && !isError;
+  const showSideBar = !isLoading && !isError;
+
+  const setSearchWith = (params: any) => {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  };
+
+  const handleQueryChange = (selectQuery: string) => {
+    setSearchWith({ query: selectQuery });
+  };
+
+  const handleSexChange = (selectSex: string | null) => {
+    setSearchWith({ sex: selectSex });
+  };
+
+  const toggleCenturies = (selectCentury: string) => {
+    return centuries.includes(selectCentury)
+      ? centuries.filter(century => century !== selectCentury)
+      : [...centuries, selectCentury];
+  };
+
+  const handleSortChange = (sortBy: string) => {
+    const isSelected = sortBy === sort;
+
+    if ((!sort && !order) || !isSelected) {
+      return { sort: sortBy, order: null };
+    }
+
+    if (sort && !order && isSelected) {
+      return { sort: sortBy, order: DESCENDING_ORDER };
+    }
+
+    if (sort && order) {
+      return { sort: null, order: null };
+    }
+
+    return { sort: null, order: null };
+  };
+
+  const preparedPeople = getPreparedPeople(people);
+
+  const getSortedPeople = () => {
+    let sortedPeople = [...preparedPeople];
+
+    if (sort) {
+      const normalizedSort = sort.charAt(0).toUpperCase() + sort.slice(1);
+
+      switch (normalizedSort) {
+        case TitleTableHeaders.name:
+          sortedPeople = sortedPeople.sort((person1, person2) =>
+            person1.name.localeCompare(person2.name),
+          );
+          break;
+        case TitleTableHeaders.sex:
+          sortedPeople = sortedPeople.sort((person1, person2) =>
+            person1.sex.localeCompare(person2.sex),
+          );
+          break;
```

Comment:
```suggestion
        case TitleTableHeaders.name:
        case TitleTableHeaders.sex:
          sortedPeople = sortedPeople.sort((person1, person2) =>
            person1[sort].localeCompare(person2[sort]),
          );
          break;
```
we can group some cases